src/luxonis_ml/data/dataset.py

```diff
@@ -658,33 +651,50 @@ def make_splits(
                 N = len(ids)
                 b1 = round(N * ratios[0])
                 b2 = round(N * ratios[0]) + round(N * ratios[1])
-                splits["train"] = ids[:b1]
-                splits["val"] = ids[b1:b2]
-                splits["test"] = ids[b2:]
+                new_splits["train"] = ids[:b1]
+                new_splits["val"] = ids[b1:b2]
+                new_splits["test"] = ids[b2:]
+                splits_to_update = ["train", "val", "test"]
             else:  # provided split
                 index = self._get_file_index()
                 if index is None:
                     raise Exception("File index not found")
-                if set(definitions.keys()) != {"train", "val", "test"}:
-                    raise Exception(
-                        "Must specify train, val, and test and those keys only"
-                    )
                 for split in "train", "val", "test":
+                    if split not in definitions:
+                        continue
```

Comment:
What is the case where we wouldn't have a split here? I'm not sure if it's something we should throw an error for instead since doesn't luxonis-train expect these 3 splits?

Reply:
The case would be where we create the dataset and then down the line we only want to update "train" split. In this case `definitions` can only be `{train:[....]}` and then only this part of the splits.json will be updated (previously definitions had to have all three keys and whole splits.json was overwritten). This then also matches with usecase of `.parse_raw_dir()` function